apps/examples/testcase/le_tc/kernel/itc_pthread.c

```diff
@@ -0,0 +1,535 @@
+static pthread_key_t g_key;
+
+/**
+* @fn                   :thread_setgetname
+* @brief                :utility function for creating thread
+* @return               :void*
+*/
+static void *thread_setgetname(void *param)
+{
+	while (1) {
+		sleep(SEC_1);
+	}
+	return NULL;
+}
+
+/**
+* @fn                   :thread_equal_check
+* @brief                :utility function for itc_pthread_pthread_equal_p
+* @return               :void*
+*/
+static void *thread_equal_check(void *param)
+{
+	sleep(SEC_1);
+	if (pthread_equal(g_pthread, pthread_self()) == false) {
+		g_flag = false;
+		return NULL;
+	}
+	g_flag = true;
+	return NULL;
+}
+
+/**
+* @fn                   :thread_set_get_specific
+* @brief                :utility function for itc_pthread_key_create_set_getspecific_p_multitime
+* @return               :void*
+*/
+static void *thread_set_get_specific(void *param)
+{
+	int *get_spec;
+	int ret_chk;
+	int index;
+	g_flag = false;
+
+	for (index = 1; index <= LOOP_CNT; index++) {
+		ret_chk = pthread_setspecific(g_key, &index);
+		if (ret_chk != OK) {
+			printf("itc_pthread_key_create_set_getspecific_p_multitime: TC FAIL\n");
+			return NULL;
+		}
+
+		get_spec = pthread_getspecific(g_key);
+		if (*get_spec != index) {
+			printf("itc_pthread_key_create_set_getspecific_p_multitime: TC FAIL\n");
+			return NULL;
+		}
+	}
+	sleep(SEC_1);
+	g_flag = true;
+	return NULL;
+}
+
+/**
+* @fn                   :task_exit
+* @brief                :utility function for itc_pthread_pthread_kill_n_rekill
+* @return               :void*
+*/
+static void *task_exit(void *param)
+{
+	pthread_setname_np(0, "task_exit");
+	pthread_exit(RETURN_PTHREAD_JOIN);
+	return NULL;
+}
+
+/**
+* @fn                   :threadfunc_sched
+* @brief                :utility function for itc_pthread_pthread_set_get_schedparam_n
+* @return               :void*
+*/
+static void *threadfunc_sched(void *param)
+{
+	pthread_setname_np(0, "threadfunc_sched");
+	sleep(SEC_8);
+	pthread_exit((pthread_addr_t)1);
+	return NULL;
+}
+
+/****************************************************************************
+  * Private Functions
+ ****************************************************************************/
+
+/**
+* @fn                   :itc_pthread_pthread_barrier_destroy_n_redestory
+* @brief                :The pthread_barrier_init() function allocates any resources required to use
+*                        the barrier referenced by 'barrier' and initialized the barrier with the
+*                        attributes referenced by attr.
+*                        barrier_wait synchronizse participating threads at the barrier referenced by 'barrier'.
+* @Scenario             :pthread_barrier_destroy re-destroying the object as already being destroyed
+* API's covered         :pthread_barrier_init, pthread_barrier_wait , pthread_barrier_destroy
+* Preconditions         :none
+* Postconditions        :none
+* @return               :void
```

Comment:
remove return type from function description. Please check for other testcases also.